src/com/xilinx/rapidwright/rwroute/RouteNode.java

```diff
@@ -169,13 +174,43 @@ private void setBaseCost() {
                         // In case of U-turn nodes
                         if (length != 0) baseCost = 0.15f * length;// VQUADs have length 4 and 5
                         break;
+                    case NODE_HLONG6:  // Versal only
+                    case NODE_HLONG10: // Versal only
+                        short minLength = length == 0 ? 1 : length;
+                        baseCost = 0.15f * minLength;
+                        break;
                     case NODE_HLONG:
                         assert (length != 0 || getAllDownhillNodes().isEmpty());
                         baseCost = 0.15f * length;// HLONGs have length 6 and 7
                         break;
+                    case NODE_VLONG7:  // Versal only
+                    case NODE_VLONG12: // Versal only
                     case NODE_VLONG:
                         baseCost = 0.7f;
```

Comment:
Should VLONG7 and VLONG12 have the same base cost?

Reply:
Seems that assign the base cost according to the length of a node is better. Has updated this part in https://github.com/Xilinx/RapidWright/pull/1077/commits/06658011533dd7ff8f2ce07d97c52541d2322932